lib/cn/omf.h

```diff
@@ -1,13 +1,160 @@
+OMF_SETGET(struct wbt_node_hdr_omf, wbn_kmd, 32)
+OMF_SETGET(struct wbt_node_hdr_omf, wbn_pfx_len, 16)
+
+/* WBT internal node entry (v6) */
+struct wbt_ine_omf {
+    uint16_t ine_koff;       /* byte offset from start of node to key */
+    uint16_t ine_left_child; /* node number of left child */
+} HSE_PACKED;
+
+OMF_SETGET(struct wbt_ine_omf, ine_koff, 16)
+OMF_SETGET(struct wbt_ine_omf, ine_left_child, 16)
+
+/* WBT leaf node entry (v6)
+ * Note, if lfe_kmd == U16_MAX, then the actual kmd offset is stored as a LE32
+ * value at lfe_koff, and the actual key is stored at lfe_koff + 4.
+ */
+struct wbt_lfe_omf {
+    uint16_t lfe_koff;
+    uint16_t lfe_kmd;
+} HSE_PACKED;
+
+OMF_SETGET(struct wbt_lfe_omf, lfe_koff, 16)
+OMF_SETGET(struct wbt_lfe_omf, lfe_kmd, 16)
+
+/*****************************************************************
+ *
+ * Hblock header OMF
+ *
+ ****************************************************************/
+
+#define HBLOCK_HDR_MAGIC UINT32_C(0xcafedead)
+
+struct hblock_hdr_omf {
+    /* integrity check, version and type */
+    uint32_t hbh_magic;
+    uint32_t hbh_version;
+
+    uint64_t hbh_min_seqno;
+    uint64_t hbh_max_seqno;
+
+    uint32_t hbh_num_kblocks;
+    uint32_t hbh_num_vblocks;
+    uint32_t hbh_num_vgroups;
+
+    /* HyperLogLog */
+    uint32_t hbh_hlog_off_pg;
+    uint32_t hbh_hlog_len_pg;
+
+    /* prefix tombstone tree */
+    struct wbt_hdr_omf hbh_ptree_hdr;
+    uint32_t hbh_ptree_data_off_pg;
+    uint32_t hbh_ptree_data_len_pg;
+
+    /* vblock index adjust */
+    uint32_t hbh_vblk_idx_adj_off_pg;
+    uint32_t hbh_vblk_idx_adj_len_pg;
+
+    /* min prefix */
+    uint32_t hbh_min_pfx_off;
+    uint8_t hbh_min_pfx_len;
+
+    /* max prefix */
+    uint32_t hbh_max_pfx_off;
+    uint8_t hbh_max_pfx_len;
+} HSE_PACKED;
+
+OMF_SETGET(struct hblock_hdr_omf, hbh_magic, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_version, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_min_seqno, 64)
+OMF_SETGET(struct hblock_hdr_omf, hbh_max_seqno, 64)
+OMF_SETGET(struct hblock_hdr_omf, hbh_num_kblocks, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_num_vblocks, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_num_vgroups, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_hlog_off_pg, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_hlog_len_pg, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_ptree_data_off_pg, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_ptree_data_len_pg, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_vblk_idx_adj_off_pg, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_vblk_idx_adj_len_pg, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_min_pfx_off, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_min_pfx_len, 8)
+OMF_SETGET(struct hblock_hdr_omf, hbh_max_pfx_off, 32)
+OMF_SETGET(struct hblock_hdr_omf, hbh_max_pfx_len, 8)
+
+static_assert(HSE_KVS_PFX_LEN_MAX <= UINT8_MAX,
+    "uint8_t is not enough to hold HSE_KVS_PFX_LEN_MAX");
+
+#define HBLOCK_HDR_PAGES ((sizeof(struct hblock_hdr_omf) + PAGE_SIZE - 1) / PAGE_SIZE)
```

Comment:
Seems like we need to add 2 * HSE_KVS_PFX_LEN_MAX in the numerator here?

Reply:
Yep we could. I will go ahead and do that. Good change.

Reply:
I would define as PAGE_SIZE and use a static assert to catch if/when it ever exceeds the page size.  As is, it could sneak above PAGE_SIZE and we wouldn't notice.